### PR TITLE
Break tests into different groups

### PR DESCRIPTION
**Pull request by devmotion**
Fixes https://github.com/JuliaGaussianProcesses/KernelFunctions.jl/issues/315.

Probably it can/should be optimized what groups and in particular how many of them we use.

### CODE REVIEW

**Comment by devmotion** [comment]
Maybe we could also drop tests on MacOS and Windows? There are no OS-specific features in KernelFunctions, so I don't think the additional tests provide any additional information. And if there are actually OS-specific bug reports at some we can still reevaluate.

**Comment by willtebbutt** [comment]
> Maybe we could also drop tests on MacOS and Windows?

I am in favour of this.

**Review by st-- — APPROVED** [review]
Looks good to me. Just a thought, to minimise the risk of someone at some point introducing a typo in `"All"`, how about changing it to `GROUP = get(ENV, "GROUP", "")` and then simply testing `if GROUP == "" || ...`? The intent is the same ("" == "no specific group").

File: test/runtests.jl [hunk 1]
```diff
@@ -19,6 +19,8 @@ using KernelFunctions: SimpleKernel, metric, kappa, ColVecs, RowVecs, TestUtils
 
 using KernelFunctions.TestUtils: test_interface
 
+const GROUP = get(ENV, "GROUP", "All")
```
Comment (st--) [inline test/runtests.jl:22]: I can't comment on the test-explanation below, but I'd add a comment along the lines of
```suggestion
const GROUP = get(ENV, "GROUP", "All")
# The GROUP is used to run different sets of tests in parallel on the GitHub Actions CI. If you want to introduce a new group, ensure you also add it to .github/workflows/ci.yml
```

File: test/runtests.jl [hunk 2]
```diff
@@ -54,123 +56,135 @@ using KernelFunctions.TestUtils: test_interface
...
+            include(joinpath("distances", "dotproduct.jl"))
+            include(joinpath("distances", "delta.jl"))
+            include(joinpath("distances", "sinus.jl"))
+        end
+        @info "Ran tests on Distances"
 
-    @testset "approximations" begin
-        include(joinpath("approximations", "nystrom.jl"))
-    end
+        if GROUP == "All" || GROUP == "Matrix"
```
Comment (st--) [inline test/runtests.jl:151]: This doesn't seem to make sense; to execute this line, GROUP can never be Matrix

Comment (devmotion) [reply test/runtests.jl:151]: Oh I forgot to remove this line when I merged the two groups :grimacing:

**Review by st-- — APPROVED** [review]

**Comment by devmotion** [comment]
@st-- I noticed that the coveralls requires some explicit handling of the parallel runs, otherwise the coverage reports are completely messed up. I just updated the action according to https://github.com/coverallsapp/github-action#complete-parallel-job-example. Can you re-approve if you are happy with the PR?

**Review by st-- — APPROVED** [review]
I'm happy with it but there still seem to be some test issues...

**Comment by devmotion** [comment]
The test error with the Julia nightly is caused by https://github.com/JuliaArrays/AxisArrays.jl/issues/200 and not related to KernelFunctions.

**Comment by st--** [comment]
@devmotion why would coverage have decreased? does this mean some tests got missed out in the split ?

**Comment by devmotion** [comment]
I don't think the drop is real, I can't find any coverage changes in the more detailed output of coveralls. I assume it is just related to the change from non-parallel to parallel tests and uploads.

**Pull request merged**